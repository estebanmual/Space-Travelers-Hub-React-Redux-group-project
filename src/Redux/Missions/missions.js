import { FETCH_MISSIONS, JOIN_MISSION, LEAVE_MISSION } from '../../constants';

// Action Creators

export const fetchMissions = () => async (dispatch) => {
  await fetch('https://api.spacexdata.com/v3/missions')
    .then((res) => res.json())
    .then((data) => dispatch({ type: FETCH_MISSIONS, missions: data }));
};

export const joinMission = (id) => ({ type: JOIN_MISSION, id });

export const leaveMission = (id) => ({ type: LEAVE_MISSION, id });

// Reducer

export default function missionsReducer(state = [], action) {
  let missions = [];

  switch (action.type) {
    case FETCH_MISSIONS:
      action.missions.forEach((mission) => {
        missions.push({
          id: mission.mission_id,
          name: mission.mission_name,
          description: mission.description,
          reserved: false,
        });
      });
      return missions;
    case JOIN_MISSION:
      missions = state.map((mission) => {
        if (mission.id !== action.id) { return mission; }
        return { ...mission, reserved: true };
      });
      return missions;
    case LEAVE_MISSION:
      missions = state.map((mission) => {
        if (mission.id !== action.id) { return mission; }
        return { ...mission, reserved: false };
      });
      return missions;
    default:
      return state;
  }
}
