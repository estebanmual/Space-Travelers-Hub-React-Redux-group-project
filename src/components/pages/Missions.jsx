import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchMissions } from '../../Redux/Missions/missions';

function Missions() {
  const missions = useSelector((state) => (state.missions));
  const dispatch = useDispatch();

  console.log(missions);

  useEffect(() => {
    dispatch(fetchMissions());
  }, []);

  return <div>Missions</div>;
}

export default Missions;
