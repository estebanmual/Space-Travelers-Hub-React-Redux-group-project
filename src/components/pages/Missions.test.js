import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../Redux/storeSetUp';
import Missions from './Missions';

describe('testing Header', () => {
  it('snapshot for the header ', () => {
    const tree = renderer.create(<Provider store={store}><Missions /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('testing Header component renders correctly', () => {
    render(<Provider store={store}><Missions /></Provider>);
    expect(screen.getByText('Mission')).toBeInTheDocument();
  });
});
