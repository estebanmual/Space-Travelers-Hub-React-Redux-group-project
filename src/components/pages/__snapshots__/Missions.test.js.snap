// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Header snapshot for the header  1`] = `
<table
  className="table table-striped table-bordered"
>
  <thead
    className="Table__Head"
  >
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
